rotkehlchen/chain/ethereum/modules/lido/decoder.py

```diff
@@ -0,0 +1,156 @@
+from rotkehlchen.constants.assets import A_ETH, A_STETH
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import ChecksumEvmAddress
+from rotkehlchen.utils.misc import hex_or_bytes_to_address, hex_or_bytes_to_int
+
+from .constants import CPT_LIDO_ETH, LIDO_STETH_SUBMITTED, STETH_MAX_ROUND_ERROR_WEI
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.user_messages import MessagesAggregator
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class LidoDecoder(DecoderInterface):
+
+    def __init__(
+            self,
+            evm_inquirer: 'EvmNodeInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+    ) -> None:
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.steth_evm_address = A_STETH.resolve_to_evm_token().evm_address
+
+    def _decode_lido_staking_in_steth(self, context: DecoderContext) -> DecodingOutput:
+        """Decode the submit of eth to lido contract for obtaining steth in return"""
+        sender_address = hex_or_bytes_to_address(context.tx_log.topics[1])
+        amount_raw = hex_or_bytes_to_int(context.tx_log.data[:32])
+        collateral_amount = token_normalized_value_decimals(
+            token_amount=amount_raw,
+            token_decimals=18,
+        )
+
+        # Searching for the exact paired stETH token reception for validating the decoding
+        steth_minted_tokens = None
+        # Here searching for the paired stETH mint operation, with a hacky way for the issue:
+        # https://github.com/lidofinance/lido-dao/issues/442
+        for tx_log in context.all_logs:
+            if (
+                tx_log.address == self.steth_evm_address and
+                tx_log.topics[0] == ERC20_OR_ERC721_TRANSFER and
+                hex_or_bytes_to_address(tx_log.topics[1]) == ZERO_ADDRESS and  # from
+                hex_or_bytes_to_address(tx_log.topics[2]) == sender_address and  # to
+                abs(amount_raw - hex_or_bytes_to_int(tx_log.data[:32])) < STETH_MAX_ROUND_ERROR_WEI
+            ):
+                steth_minted_tokens = token_normalized_value_decimals(
+                    token_amount=hex_or_bytes_to_int(tx_log.data[:32]),
+                    token_decimals=18,
+                )
+                break
+        else:  # did not break/find anything
+            log.error(
+                f'At lido steth submit decoding of tx {context.transaction.tx_hash.hex()}'
+                f'did not find the related stETH token generation',
+            )
+            return DEFAULT_DECODING_OUTPUT
```

Comment:
So I had an idea here to improve this approach.

I think you don't need to do this search of all logs. Instead since the ActionItem will already do this matching for you for everything **except** this `STETH_MAX_ROUND_ERROR_WEI` why not move this to the `ActionItem` class itself? So have an extra optional attribute. Something like `amount_error_tolerance` in the `ActionItem` class? 

And here:

https://github.com/rotki/rotki/blob/664484982a3b2eb23f9afbe2e12688d67e153840/rotkehlchen/chain/evm/decoding/decoder.py#L906

modify the check if the `action_item.amount_error_tolerance is not None` to do the absolute check you have there.

So you keep what you have here except this check of all logs. Does this make sense?

```suggestion
```